src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -68,12 +70,30 @@ public class DedupManifestArtifactClientFactory : IDedupManifestArtifactClientFa
         // At 192x it was around 16 seconds and 256x was no faster.
         private const int DefaultDedupStoreClientMaxParallelism = 192;
 
-        public static readonly DedupManifestArtifactClientFactory Instance = new DedupManifestArtifactClientFactory();
+        public Microsoft.VisualStudio.Services.BlobStore.WebApi.Contracts.Client? Client { get; }
 
-        private DedupManifestArtifactClientFactory()
+        public HashType? HashType { get; private set; }
+
+        private static DedupManifestArtifactClientFactory _instance;
+
+        public static DedupManifestArtifactClientFactory Instance => _instance ?? throw new InvalidOperationException($"Initialize the instance before using.");
+
+        public static void Initialize(Microsoft.VisualStudio.Services.BlobStore.WebApi.Contracts.Client client, HashType? hashType)
```

Comment:
This method is called a lot in this PR and it looks like if it is called twice it will throw an exception. How do we assure that this is never called twice? Also having this static seems like an anti-pattern, is it necessary? should we be adding HashType to the factory constructor rather than having it be a static?